### ISSUE TITLE
Loaded but Failed to Enable v2.0.1

### ISSUE THREAD

**Issue by davebhc71** [delashum/obsidian-checklist-plugin#110]
Loaded but Failed to Enable v2.0.1 On iPad.
Obsidian Mobile v1.05

Used previous versions without issues. Closed Obsidian, restarted. Same error. Uninstalled plugin. Closed Obsidian, restarted Obsidian, reinstalled plugin, same error when trying to enable. Nothing else has changed on system. Bummer!

**Comment by JamesHWade** [comment]
I am seeing the same thing. Desktop is still working as expected.

**Comment by NavinChen** [comment]
me too……
mobile version not work

**Comment by delashum** [comment]
😔 i'll try to look this weekend.

**Comment by NavinChen** [comment]
> 😔 i'll try to look this weekend.

Thanks for your hard work

**Comment by JamesHWade** [comment]
Yes, thank you for your great work on this plugin!

**Comment by Broodje-met-Beleg** [comment]
I have the same problem. Looking forward to using it on the iPad as well.

**Comment by stonesphinx** [comment]
Confirming error 'Failed to load plugin obsidian-checklist-plugin' whenever I try to enable it. 
Android Galaxy S20
Obsidian 1.0.5. 

Tried reinstalling Checklist, restarting obsidian, etc. Tried rolling back to 2.0.0. Nothing. 

Great plugin - I hope you can get it working again!

**Comment by delashum** [comment]
Can anyone in this thread tell me the error you're seeing in the console? I'm having a hard time reproducing this

**Comment by takato3000** [comment]
It seems that "Include Files" become glob pattern.
for example `(Periodic Notes)/**/*` includes all files in Periodic Notes directory.

**Comment by stonesphinx** [comment]
I tried to getting[ debugging mode](https://forum.obsidian.md/t/debugging-obsidian-mobile-plugins/20913) enabled so I could view the the console and see what errors are throwing, but my device never showed up in Chrome when following the directions. 

I'm not a developer or programmer in any way. 

If there's another way to get access to the console to see error codes on Mobile I want to contribute, since I'm invested in getting this plugin working again. How-to instructions would have to be suitable to a five year old so I can follow along.

**Comment by delashum** [comment]
Yes, v2 came with some breaking changes, one of them being that include files is not glob style pattern matching. Did that resolve this issue for you somehow @takato3000 

@stonesphinx thanks! is `sep` a tag you use? Unfortunately since it's minified those logs don't give too much info besides that.

UPDATE: I can still reproduce this on my iPhone but when using on my desktop with `mobileEmulation(true)` it works fine, so having a hard time finding the root cause here.

**Comment by stonesphinx** [comment]
Nope, not sure what `sep` could be. The only tag I used in Checklist is #ingredients 

Anything else I can do to give you more information?

**Comment by OllieA** [comment]
Updated to 2.0.2 hoping it would fix this issue as well as the formatting one. Unfortunately still getting “failed to load plugin” when try to enable checklist on both iPhone and iPad.

**Comment by delashum** [comment]
`2.0.2` fixed a different problem, which is why this isn't closed yet. Need to try and get access to the error logs on mobile to see what's going on

**Comment by takato3000** [comment]
@delashum After I upgraded to 2.0.2, it became loading icon again.
At this time, when I use `**/*`, it's still always loading.
But `(somedirectory)/**/*` pattern still works on my friend's computer.(2.0.2)

That's strange.

**Comment by stonesphinx** [comment]
Update: The `Failed to load plugin obsidian-checklist-plugin` persists after the 2.0.2 upgrade. I tried a few more things: reinstalling Obsidian on my phone, installing the plugin on a different vault (thinking perhaps there was a plugin or vault-specific conflict), etc. I continue to get the same error.

An interesting note is that (If I remember correctly) the plugin was working for several days post 2.0.x upgrade, and then stopped. 

The only thing that worked is rolling back the plugin to 1.2.3, which allows it to load on mobile again. 

(This whole time 2.0.x has functioned normally on both windows 11 PCs where my vault is synced.)

**Comment by davebhc71** [comment]
“Need to try and get access to the error logs on mobile to see what's going on“

Any progress?

**Comment by AndrewNatoli** [comment]
Hey @stonesphinx - Are you able to share which method you used for getting access to the console and if that was on iOS? I tried out [Liam Cain's snippet](https://gist.github.com/liamcain/3f21f1ee820cb30f18050d2f3ad85f3f) (tossing `monkeyPatchConsole(this)` into the beginning of `onLoad`) but it's not generating an output file for me on mobile; just desktop if I take that platform restriction line off.

**Comment by AndrewNatoli** [comment]
I may have found where `sep` comes from. In the minified JS (`npm run build`), `.sep` is used in something to deal with filesystem paths and comes from the CommonJS resolver.

I tried updating the rollup CommonJS plugin.... it changed the output code but didn't fix the problem on mobile. Tried upgrading rollup and _all_ node modules to their latest versions-- still have the error. Can't get console access so I'm not sure if the error changed, but the `.sep` reference still exists. I checked out [Obsidian Dataview](https://github.com/blacksmithgu/obsidian-dataview) which also uses rollup but runs successfully on mobile-- the minified output doesn't contain the same `.sep` code.

So it _might_ be something with the rollup config...

**Comment by stonesphinx** [comment]
@AndrewNatoli I followed [these directions](https://forum.obsidian.md/t/debugging-obsidian-mobile-plugins/20913) to get access to the mobile console on my windows PC. My android phone was plugged in via USB to make this work. 

Glad to see you're making progress - if you want to commit your latest changes (which is throwing the the latest error), I can see if it gives the same error on mine.

**Comment by dammitcoetzee** [comment]
Any word on this? would love to use this on mobile again.

**Comment by NomarCub** [comment]
I've had to switch to the Dataview plugin unfortunately. It has the functionality that I wanted out of this plugin, and is actively maintained. However they are planning major changes to the task view there (which already went live a few weeks ago but got rolled back).

**Comment by dmkravets** [comment]
I'm just checking in to see if the mobile version will get updated so it will load? 🙏🏼 thanks

**Comment by jconti** [comment]
The locked in versions of roll up dependencies look old. Maybe previous Obsidian builds included a dependency that was omitted in 2.0.2? There could easily be fixed tree-shaking bugs in rollup. An update might re-include any missing code.

**Comment by flankstaek** [comment]
Please download the latest update and try again.

**Comment by stonesphinx** [comment]
I updated the plugin to 2.2.3 and it appears to be working on mobile!

**Comment by dammitcoetzee** [comment]
it works! thanks so much!
